src/main/kotlin/com/exactpro/th2/codec/oracle/logminer/LogMinerTransformer.kt

```diff
@@ -164,31 +586,146 @@ class LogMinerTransformer(private val config: LogMinerConfiguration) : IPipeline
             setProtocol(this@toBuilderWithoutBody.protocol)
         }
 
-        internal fun Function.toReadable(onWarning: (String) -> Unit): Map<String, Any?> {
-            return hashMapOf(
-                "function" to name,
-                "parameters" to parameters.map { it.toReadable(onWarning) }
-            )
+        private fun Throwable.extractMessage(): String? = message?.lines()?.first()
+
+        private inline fun <T> MutableList<T>.removeLastWhile(predicate: (T) -> Boolean): List<T> {
+            if (isEmpty())
+                return emptyList()
+            val iterator = listIterator(size)
+            while (iterator.hasPrevious()) {
+                if (!predicate(iterator.previous())) {
+                    iterator.next()
+                    val expectedSize = size - iterator.nextIndex()
+                    if (expectedSize == 0) return emptyList()
+                    return ArrayList<T>(expectedSize).apply {
+                        while (iterator.hasNext()) {
+                            add(iterator.next())
+                            iterator.remove()
+                        }
+                    }
+                }
+            }
+            return toList()
         }
 
-        internal fun Expression.toReadable(onWarning: (String) -> Unit): Any? {
-            return when (this) {
-                is Function -> toReadable(onWarning)
-                is StringValue -> value.trim()
-                is LongValue -> value
-                is NullValue -> null
-                else -> {
-                    onWarning("Unsupported expression: '$this', type: ${this::class.java}")
-                    toString()
+        private interface Expression {
+            val value: Any?
+            val tokenIndex: Int
+            val isCompleted: Boolean
+
+            fun apply(value: Any?)
```

Comment:
We should avoid using methods with names that intersect with well-known methods from the Kotlin std library. It might confuse people when they see the well-known method name but this method behaves differently